enhancements/local-storage/health-monitoring.md

```diff
@@ -0,0 +1,137 @@
+- As an admin, I would like if components (such as OpenShift Data Foundation) could access storage device health and failover faster.
+
+
+### Implementation Details/Notes/Constraints
+
+`LocalVolumeDisocveryResults` already enumerate and refresh information about the disk in a per-node `LocalVolumeDiscoveryResult`, and a [PR](https://github.com/openshift/local-storage-operator/pull/249) is in progress to add metrics to each of the daemonsets that update the CR.
+
+`LocalVolumeDiscovery` can be configured via `nodeSelector` and `toleration` to match any set of nodes or all nodes. We could have the metrics be for all nodes or just for the ones that LocalVolumeDiscovery is configured to watch. Whether this should be controlled by the same (singleton) CR is an [open question](#Open-Questions    )
+
+We will be collecting metrics via [libstoragemgmt](https://github.com/libstorage/libstoragemgmt) and calling it via [libstoragemgmt-golang](https://github.com/libstorage/libstoragemgmt-golang/).
+If a block device is a partion or logical volume, we will only expose health for the parent device.
+
+
+### Risks and Mitigations
+
+- There is some concern that this is out of scope for Local Storage Operator
+  - We are discussing [alternatives](#Alternatives).
+- There is some uncertainty about whether this implementation precludes other components correlating PersistentVolumes with this information.
+  - Mitigation: We are investigating if it is an acceptable pattern for internal components to scrape and correlate cluster prometheus metrics for automation. There has been some indication that this is the case on #forum-monitoring.
+- The golang bindings for libstoragemgmgt also needs some review and polish.
+  - Mitigation: The ODF(OpenShift Data Foundation) team will contribute to and help the RHEL team maintain libstoragemgmt golang
+
+
+## Design Details
+
+### Open Questions
+
+- Is Local Storage Operator the best place for this?
+  - see [alternatives](#Alternatives)
+- Should we allow configuration to watch only a subset of nodes or watch all nodes by default? Should the metrics be controller by LocalVolumeDiscovery (singleton)
+
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+- E2E tests will deploy the component and verify that metrics are being emitted per device
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+
+
+### Upgrade / Downgrade Strategy
+
+
+
+### Version Skew Strategy
+
+
+
+## Implementation History
+
+
+## Drawbacks
+
+
+
+## Alternatives
+
+- We *could* create a new component for this.
+
+Disqualified alternatives:
+- We explored node_exporter, but it is a rootless container, so SMART data can't be checked.
```

Comment:
Why can't node exporter be made to run as root?

Reply:
My understanding is that upstream won't go for it based on [this issue](https://github.com/prometheus/node_exporter/issues/782#issuecomment-356160910) and asking around on #forum-monitoring.

Reply:
Why can't OpenShift run it as root?

Reply:
Running this as privileged is not a problem in OCP. We are already running some collectors in privileged containers as part of same node-exporter pod.

Reply:
If that's the case, why was this approach disqualified?

Reply:
Hi @gnufied, would be useful if you could involve our team in this discussion. Can you link to the slack conversation?